### PR TITLE
zthrottle: init

### PR DESCRIPTION
**Pull request by ckiee**
###### Motivation for this change
I needed it packaged so I could use it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/zthrottle/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchFromGitHub, zsh }:
+
+stdenv.mkDerivation rec {
+  pname = "zthrottle";
+  version = "git-2017-7-24";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/zthrottle/default.nix:5]:
```suggestion
  version = "unstable-2017-7-24";
```

File: pkgs/tools/misc/zthrottle/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "anko";
+    repo = "zthrottle";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/zthrottle/default.nix:9]:
```suggestion
    repo = pname;
```
opinionated change :p

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 124523` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zthrottle</li>
  </ul>
</details>

File: pkgs/tools/misc/zthrottle/default.nix (hunk 1 above)
Comment (ronthecookie) [reply pkgs/tools/misc/zthrottle/default.nix]: Thanks for the quick review :D

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124523` at af4d20a6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>zthrottle</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/zthrottle/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/zthrottle/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  buildInputs = [ zsh ];
+
+  installPhase = ''
+    install -D zthrottle $out/bin/zthrottle
+  '';
+
+  meta = with lib; {
+    description =
+      "A program that throttles a pipeline, only letting a line through at most every $1 seconds.";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/zthrottle/default.nix:22]:
```suggestion
    description = "A program that throttles a pipeline, only letting a line through at most every $1 seconds";
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -31643,5 +31643,7 @@ in
 
   xcolor = callPackage ../tools/graphics/xcolor { };
 
+  zthrottle = callPackage ../tools/misc/zthrottle {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:31646]:
```suggestion
  zthrottle = callPackage ../tools/misc/zthrottle { };
```

File: pkgs/tools/misc/zthrottle/default.nix (hunk 3 above)
Comment (ronthecookie) [reply pkgs/tools/misc/zthrottle/default.nix]: Done even though I think this is quite silly as I was just letting `nixfmt` do the formatting work and I don't think we have a proper formatting style?

Comment (legendofmiracles) [reply pkgs/tools/misc/zthrottle/default.nix]: nixpkgs-fmt should be used in nixpkgs

Comment (ronthecookie) [reply pkgs/tools/misc/zthrottle/default.nix]: Oh I see! Will note for future PRs.

**Pull request merged**